aws/resource_aws_api_gateway2_api.go

```diff
@@ -0,0 +1,225 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/apigatewayv2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsApiGateway2Api() *schema.Resource {
```

Comment:
```suggestion
func resourceAwsApiGatewayV2Api() *schema.Resource {
```
Should this and the file itself contain `V2` (i.e. `resource_aws_api_gatewayv2_api`) to keep consistent with AWS naming?

Reply:
@kreempuff I followed the naming for CloudHSM V2 - for example [`aws_cloudhsm_v2_cluster`](https://www.terraform.io/docs/providers/aws/r/cloudhsm_v2_cluster.html) resource is implemented in file [`resource_aws_cloudhsm2_cluster.go`](https://github.com/terraform-providers/terraform-provider-aws/blob/34ff52a5834641323eb566ae731f36e6814bc268/aws/resource_aws_cloudhsm2_cluster.go) with entrypoint [`resourceAwsCloudHsm2Cluster()`](https://github.com/terraform-providers/terraform-provider-aws/blob/34ff52a5834641323eb566ae731f36e6814bc268/aws/resource_aws_cloudhsm2_cluster.go#L16).
Happy to change.

Reply:
@ewbankkit Makes sense but `CloudHSM`looks like it replaced the previous HSM service as an upgrade, whereas API Gateway and API Gateway V2 exist side by side as two similar but different services. 

I don't think it needs to be changed, but it's a little food for thought 👍 


Great job either way. Can't wait for this to be merged